src/rgw/rgw_rest_conn.cc

```diff
@@ -394,12 +469,16 @@ int RGWRESTConn::send_resource(const DoutPrefixProvider *dpp, const std::string&
   ret = req.send_request(dpp, &key, headers, resource, mgr, send_data);
   if (ret < 0) {
     ldpp_dout(dpp, 5) << __func__ << ": send_request() resource=" << resource << " returned ret=" << ret << dendl;
+    if (ret == -EIO)
+      set_url_unconnectable(url);
```

Comment:
would errors from `send_request()` ever indicate a connection error? my understanding is that just signs the request and enqueues it for sending. i think we should only handle the errors from `complete_request()`